charts/kong/templates/ingress-admin.yaml

```diff
@@ -29,4 +30,36 @@ spec:
     - {{ $hostname }}
     secretName: {{ $tls }}
   {{- end -}}
+{{- else -}} {{/* Modern admin handler */}}
+{{- $serviceName := include "kong.fullname" . -}}
+{{- $servicePort := include "kong.ingress.servicePort" .Values.admin -}}
+{{- $path := .Values.admin.ingress.path -}}
+{{- $tls := .Values.admin.ingress.tls -}}
+{{- $hostname := .Values.admin.ingress.hostname -}}
```

Comment:
Let's not repeat these common properties here. Please move them under the main `if` block on line 1 instead, and re-use the variables for both the cases.

Reply:
Ditto comment from the service. The duplication is by design to make cutting out the legacy block easier later.